src/coverlet.msbuild.tasks/CoverageResultTask.cs

```diff
@@ -75,13 +75,14 @@ public override bool Execute()
             {
                 Console.WriteLine("\nCalculating coverage result...");
 
-                if (InstrumenterState is null || !File.Exists(InstrumenterState.ItemSpec))
+                IFileSystem fileSystem = (IFileSystem)DependencyInjection.Current.GetService(typeof(IFileSystem));
+                if (InstrumenterState is null || !fileSystem.Exists(InstrumenterState.ItemSpec))
                 {
                     _logger.LogError("Result of instrumentation task not found");
                     return false;
                 }
 
-                var coverage = new Coverage(CoveragePrepareResult.Deserialize(new FileStream(InstrumenterState.ItemSpec, FileMode.Open)), this._logger, (IInstrumentationHelper)DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)));
+                var coverage = new Coverage(CoveragePrepareResult.Deserialize(new FileStream(InstrumenterState.ItemSpec, FileMode.Open)), this._logger, fileSystem, (IInstrumentationHelper)DependencyInjection.Current.GetService(typeof(IInstrumentationHelper)));
```

Comment:
also for `new FileStream(InstrumenterState.ItemSpec, FileMode.Open)` and surround by using please(my past error)

Reply:
Ok I changed it but I don't know how to surround this by a using.

Reply:
something like
```cs
Coverage coverage = null;
using(Stream instrumenterStateStream = fileSystem.NewFileStream(InstrumenterState.ItemSpec,FileMode.Open))
{
     coverage = new Coverage(CoveragePrepareResult.Deserialize(instrumenterStateStream), this._logger, fileSystem,...
}
```

Reply:
Done